diff --git a/include/string.h b/include/string.h
index 795a2abc..5f3e92c2 100644
--- a/include/string.h
+++ b/include/string.h
@@ -93,9 +93,6 @@ char *strcasestr(const char *, const char *);
 void *memmem(const void *, size_t, const void *, size_t);
 void *memrchr(const void *, int, size_t);
 void *mempcpy(void *, const void *, size_t);
-#ifndef __cplusplus
-char *basename();
-#endif
 #endif
 
 #ifdef __cplusplus
